sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -472,6 +472,19 @@ object Overlay {
     builder.append(input.substringSQL(pos + length, Int.MaxValue))
     builder.build()
   }
+
+  def calculate(input: Array[Byte], replace: Array[Byte], pos: Int, len: Int): Array[Byte] = {
+    // If you specify length, it must be a positive whole number or zero.
+    // Otherwise it will be ignored.
+    // The default value for length is the length of replace.
+    val length = if (len >= 0) {
+      len
+    } else {
+      replace.length
+    }
+    ByteArray.concat(ByteArray.subStringSQL(input, 1, pos - 1),
+      replace, ByteArray.subStringSQL(input, pos + length, Int.MaxValue))
+  }
```

Comment:
Can't we expand this logic into generated code directly instead of calling this helper function?

Reply:
I am not very good at using generated code. If this is a strong suggestion, I will try to use generated code. But I think this helper function is OK too.

Reply:
cc: @viirya @mgaido91

Reply:
we are already doing the same for the string case. I don't see strong reasons for inlining this in the generated code

Reply:
ok, thanks for the check, @mgaido91

Reply:
Seems good to me too.